### PR TITLE
Remove content-decoder

### PR DESCRIPTION
**Pull request by tusharmath**
Should close — #570 #552

### CODE REVIEW

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 17.04**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 431276.34**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 416915.09**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 418858.67**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 424286.44**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 428684.73**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 421530.35**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 426516.19**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 438531.96**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 435074.34**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 423398.01**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 437061.40**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 431046.86**

File: zio-http/src/main/scala/zhttp/service/Handler.scala [hunk 1]
```diff
@@ -1,150 +1,54 @@
...
   runtime: HttpRuntime[R],
   config: Server.Config[R, Throwable],
   serverTime: ServerTimeGenerator,
-) extends ChannelInboundHandlerAdapter
+) extends SimpleChannelInboundHandler[FullHttpRequest](false)
     with WebSocketUpgrade[R] { self =>
 
-  override def channelRead(ctx: ChannelHandlerContext, msg: Any): Unit = {
+  override def channelRead0(ctx: ChannelHandlerContext, jReq: FullHttpRequest): Unit = {
+    jReq.touch("server.Handler-channelRead0")
```
Comment (smehta91) [inline zio-http/src/main/scala/zhttp/service/Handler.scala:27]: Can be removed now.

File: zio-http/src/main/scala/zhttp/http/Request.scala [hunk 2]
```diff
@@ -10,31 +11,26 @@ trait Request extends HeaderExtension[Request] { self =>
...
-
-      override def getBody[R, B](
-        decoder: ContentDecoder[R, Throwable, Chunk[Byte], B],
-      ): ZIO[R, Throwable, B] =
-        self.getBody(decoder)
+      override def method: Method                     = m
+      override def url: URL                           = u
+      override def getHeaders: List[Header]           = h
+      override def remoteAddress: Option[InetAddress] = self.remoteAddress
+      override private[zhttp] def getBodyAsByteBuf    = self.getBodyAsByteBuf
```
Comment (smehta91) [inline zio-http/src/main/scala/zhttp/http/Request.scala:18]:
```suggestion
      override private[zhttp] def getBodyAsByteBuf    = self.getBodyAsByteBuf
      override private[zhttp] def getNativeRequest     = self.getNativeRequest
```

**Pull request merged**